skimage/restoration/_denoise.py

```diff
@@ -376,21 +390,50 @@ def _wavelet_threshold(img, wavelet, threshold=None, sigma=None, mode='soft'):
            DOI: 10.1093/biomet/81.3.425
 
     """
-    coeffs = pywt.wavedecn(img, wavelet=wavelet)
-    detail_coeffs = coeffs[-1]['d' * img.ndim]
+    if not isinstance(wavelet, pywt.Wavelet):
```

Comment:
Just assign, don't check - we've already specified allowed options in the docstring.